### ISSUE TITLE
Namespace

### ISSUE THREAD

**Issue by rowanc1** [simpeg/simpeg#153]
_From @lheagy on July 11, 2015 20:44_

We attach FDEM to everything (it is a bit of overkill). What do you guys think of: 

Sources: 
EM.FDEM.SrcFDEM.SrcFDEM_MagDipole --> EM.FDEM.Src.MagDipole 

Receivers:
EM.FDEM.RxFDEM --> EM.FDEM.Rx 

Problem:
EM.FDEM.ProblemFDEM_b --> EM.FDEM.Problem_b 

Fields:
EM.FDEM.FieldsFDEM_b --> EM.FDEM.Fields_b

@rowanc1 @sgkang @grosenkj  

_Copied from original issue: simpeg/simpegem#29_

**Comment by rowanc1** [comment]
_From @sgkang on July 14, 2015 9:1_

Sounds reasonable. What happen when we use same module on time domain? For example, we may share some sources in TDEM.

Cheers,
Seogi

**Comment by rowanc1** [comment]
_From @lheagy on July 14, 2015 15:38_

There is still some work to be done to figure it out for the time domain, but I think we can use the same structure / namespace, ie for a TDEM source: EM.TDEM.Src.MagDipole instead of what it is currently. (We may be able to have the TDEM and FDEM sources inherit from the same base class?)

**Comment by rowanc1** [comment]
_From @lheagy on July 16, 2015 23:15_

or do we want:

EM.FDEM.Survey.Rx
EM.FDEM.Survey.Src.MagDipole

(similar for TDEM)
?

**Comment by rowanc1** [comment]
_From @sgkang on July 17, 2015 5:57_

I prefer the past one.



2015-07-17 8:15 GMT+09:00 Lindsey <notifications@github.com>:

**Comment by rowanc1** [comment]
Yeah, I like them under something other than survey. Like your first setup.

**Comment by rowanc1** [comment]
_From @lheagy on July 17, 2015 15:39_

will do!

**Comment by rowanc1** [comment]
_From @lheagy on July 23, 2015 23:4_

Another one for you all:

There are a lot of derivatives to keep track of, but it boils down to taking derivatives wrt the model, m, and the field we compute, u. My suggestion is that any time we create a thing, ie. getA, we should specify which derivative we are taking: ie. getADeriv_m. This becomes important when taking derivatives of fields in the fields object as aliased fields have (potentially) derivatives wrt both m and u. 

So basically any time "Deriv" is used, it should be followed by either "_u" or "_m". What do you guys think?

**Comment by rowanc1** [comment]
Makes sense as we are talking about the function c(m,u). Would clear a lot
up for me in the flow code.

**Comment by lheagy** [comment]
do we want 

a) `EM.FDEM.SurveyFDEM` and `EM.TDEM.SurveyTDEM `

or 

b) `EM.FDEM.Survey` and `EM.TDEM.Survey `

I vote (b)!

**Comment by lheagy** [comment]
and `EM.FDEM.ProblemFDEM_e` or `EM.FDEM.Problem_e` 

I this case, I am fine with the first one, esp. once we start getting analytics in there, we can name them along the lines of `EM.FDEM.ProblemAnalytic_XXX`

**Comment by lheagy** [comment]
The drawback to this though is that you don't want to import Survey from SimPEG and survey from FDEM at the same time (which shouldn't be an issue in too many cases, but worth considering)

**Comment by bsmithyman** [comment]
I would vote (b); worst case, you can always `from EM.FDEM import Survey as SurveyFDEM` if you need to.

**Comment by lheagy** [comment]
Ok, I like that
